### ISSUE TITLE
Google Automatic translation endless requests

### ISSUE THREAD

**Issue by minusf** [WeblateOrg/weblate#3991]
**Describe the bug**

Automatic translation keeps spinning with the following exception:

Internal Server Error: /js/task/f3f7db6f-f4df-46c9-85e6-294598e7f855/

TypeError at /js/task/f3f7db6f-f4df-46c9-85e6-294598e7f855/
Object of type 'KeyError' is not JSON serializable

```
2020-06-04 12:25:36,718 ERROR: log.py:228: Internal Server Error: /js/task/f3f7db6f-f4df-46c9-85e6-294598e7f855/
Traceback (most recent call last):
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/weblate/trans/views/js.py", line 266, in task_progress
    'result': task.result,
 File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/http/response.py", line 561, in __init__
    data = json.dumps(data, cls=encoder, **json_dumps_params)
 File "/usr/lib64/python3.6/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib64/python3.6/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib64/python3.6/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/export/apps/weblate/venv/lib64/python3.6/site-packages/django/core/serializers/json.py", line 104, in default
    return super().default(o)
  File "/usr/lib64/python3.6/json/encoder.py", line 180, in default
    o.__class__.__name__)
TypeError: Object of type 'KeyError' is not JSON serializable
```

Dont think this is #3375, the celery job with queue `translate` is running.

Also single items are translated **correctly without an error** with "Machine Translation"
button.

Unfortunately updating to newer weblate is not an option until the database migration without database superuser makes into a release.

**To Reproduce**
1. Enable automatic translation using Google Translate.
2. See error in logs

**Expected behavior**
Automatic translation finishes.

**Server configuration and status**
Please paste the output of `list_versions` and `check --deploy` commands over
here. Depending on installation these can be executed in different way, please
consult https://docs.weblate.org/en/latest/admin/management.html for more
details.

On pip installed Weblate:

```
weblate list_versions

 * Weblate: 3.11.3
 * Django: 3.0.6
 * siphashc: 1.3
 * Whoosh: 2.7.4
 * translate-toolkit: 2.5.1
 * lxml: 4.5.1
 * Pillow: 7.0.0
 * bleach: 3.1.5
 * six: 1.14.0
 * python-dateutil: 2.8.1
 * social-auth-core: 3.2.0
 * social-auth-app-django: 3.1.0
 * django-crispy-forms: 1.8.1
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.11.0
 * django-appconf: 1.0.4
 * user-agents: 2.1
 * filelock: 3.0.12
 * setuptools: 39.2.0
 * jellyfish: 0.7.2
 * openpyxl: 3.0.3
 * celery: 4.4.2
 * kombu: 4.6.8
 * celery-batches: 0.3
 * translation-finder: 1.8
 * html2text: 2020.1.16
 * pycairo: 1.19.1
 * pygobject: 3.36.1
 * diff-match-patch: 20181111
 * requests: 2.23.0
 * django-redis: 4.11.0
 * hiredis: 1.0.1
 * sentry_sdk: 0.14.4
 * Cython: 0.29.19
 * misaka: 2.1.1
 * GitPython: 3.1.2
 * borgbackup: 1.1.11
 * Python: 3.6.8
 * Git: 2.22.2
 * psycopg2-binary: 2.8.5
 * chardet: 3.0.4
 * ruamel.yaml: 0.16.10
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: localhost
 * Celery: redis://127.0.0.1:6380, redis://127.0.0.1:6380, regular
 * Platform: Linux 3.10.0-1062.4.1.el7.x86_64 (x86_64)
```

**Comment by nijel** [comment]
Already fixed in 4.0 by e4d935c7fb3, please upgrade.

**Comment by minusf** [comment]
yeah i just noticed when i looked at the file with a newer version.  thanks.
